### PR TITLE
Navigator port

### PR DESCRIPTION
**Pull request by tumbili**

### CODE REVIEW

File: src/modules/navigator/navigator_main.cpp [hunk 1]
```diff
@@ -516,9 +519,9 @@ Navigator::start()
 	/* start the task */
 	_navigator_task = px4_task_spawn_cmd("navigator",
 					 SCHED_DEFAULT,
-					 SCHED_PRIORITY_DEFAULT + 20,
+					 SCHED_PRIORITY_MAX -5,
```
Comment (tumbili) [inline src/modules/navigator/navigator_main.cpp]: @mcharleb With the Linux defines SCHED_PRIORITY_DEFAULT + 20 becomes 119 and obviously fails.

File: src/platforms/px4_defines.h [hunk 2]
```diff
@@ -146,6 +146,8 @@ __END_DECLS
 #define OK 0
 #define ERROR -1
 
+#define MAX_RAND 32767
```
Comment (tumbili) [inline src/platforms/px4_defines.h]: @mcharleb Please move wherever you think is right.

File: src/modules/navigator/navigator_main.cpp [hunk 3]
```diff
@@ -623,15 +630,18 @@ int navigator_main(int argc, char *argv[])
 
 	} else if (!strcmp(argv[1], "status")) {
 		navigator::g_navigator->status();
+		return 0;
```
Comment (mcharleb) [inline src/modules/navigator/navigator_main.cpp]: These "return 0" calls are not needed for every else. You will go to the bottom and do a return 0.

File: src/platforms/px4_defines.h (hunk 2 above)
Comment (mcharleb) [inline src/platforms/px4_defines.h]: I think its fine to keep in px4_defines.h

**Pull request merged**